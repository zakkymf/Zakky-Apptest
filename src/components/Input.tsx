import React from 'react';
import {
  View,
  Text,
  TextInput,
  StyleSheet,
  StyleProp,
  ViewStyle,
  TextStyle,
} from 'react-native';
import {Fonts} from '../theme';

interface InputProps {
  editable?: boolean;
  value: string;
  label: string;
  placeholder?: string;
  keyboardType?: any;
  containerStyle?: StyleProp<ViewStyle>;
  textInputStyle?: StyleProp<TextStyle>;
  onChangeText?: (value: string) => void;
}

const Input: React.FC<InputProps> = ({
  value,
  label,
  editable,
  keyboardType,
  placeholder,
  onChangeText,
  textInputStyle,
  containerStyle,
}) => {
  return (
    <View style={[styles.container, containerStyle]}>
      <Text style={styles.label}>{label}</Text>
      <TextInput
        value={value}
        testID="input"
        editable={editable}
        style={[styles.input, textInputStyle]}
        placeholder={placeholder}
        onChangeText={onChangeText}
        keyboardType={keyboardType}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginBottom: 15,
    flexDirection: 'column',
  },
  label: {
    fontSize: 12,
    fontFamily: Fonts.type.monserrat,
  },
  input: {
    height: 40,
    marginTop: 5,
    width: '100%',
    fontSize: 14,
    borderBottomWidth: 0.5,
    fontFamily: Fonts.type.monserrat,
  },
});

export default Input;
