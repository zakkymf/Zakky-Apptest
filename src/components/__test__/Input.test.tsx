import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Input from '../Input';

describe('@components/Input', () => {
  it('render input component', () => {
    const component = () =>
      render(
        <Input value="" label="First Name" onChangeText={() => jest.fn()} />,
      );

    const {toJSON} = component();
    expect(toJSON()).toMatchSnapshot();
  });

  it('user input correctly', () => {
    const {queryByTestId} = render(
      <Input value="123" label="First Name" onChangeText={() => jest.fn()} />,
    );

    const input: any = queryByTestId('input');
    fireEvent.changeText(input, '123');
    expect(input.props.value).toBe('123');
  });
});
