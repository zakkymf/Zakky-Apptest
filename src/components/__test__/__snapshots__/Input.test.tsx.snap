// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/Input render input component 1`] = `
<View
  style={
    Array [
      Object {
        "flexDirection": "column",
        "marginBottom": 15,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "fontFamily": "Montserrat-Regular",
        "fontSize": 12,
      }
    }
  >
    First Name
  </Text>
  <TextInput
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "borderBottomWidth": 0.5,
          "fontFamily": "Montserrat-Regular",
          "fontSize": 14,
          "height": 40,
          "marginTop": 5,
          "width": "100%",
        },
        undefined,
      ]
    }
    testID="input"
    value=""
  />
</View>
`;
